enigma-swing/src/main/java/cuchaz/enigma/gui/elements/EditorTabbedPane.java

```diff
@@ -80,7 +81,9 @@ public void onTitleChanged(EditorPanel editor, String title) {
 
 		if (editorPanel != null && activeEditor != editorPanel) {
 			this.openFiles.setSelectedComponent(this.editors.get(entry).getUi());
-			this.gui.showStructure(editorPanel);
+			if (gui.getRightPanel().getId().equals(RightPanel.Type.STRUCTURE)) {
+				this.gui.showStructure(editorPanel);
```

Comment:
This should be invoked on any conditions to update the panel, even if it's not the current one. Otherwise, changing the selected class when it isn't open keeps the structure of the previous class.
Following that, the method should be renamed or moved to `updateStructure`